dbms/src/main/java/org/polypheny/db/processing/AbstractQueryProcessor.java

```diff
@@ -1521,4 +1521,24 @@ private CachedProposedRoutingPlan selectCachedPlan( List<CachedProposedRoutingPl
         return (CachedProposedRoutingPlan) routingPlan;
     }
 
+
+    @Override
+    public void unlock( Statement statement ) {
+        LockManager.INSTANCE.unlock( LockManager.GLOBAL_LOCK, (TransactionImpl) statement.getTransaction() );
+    }
+
+
+    /**
+     * To acquire a global shared lock for a statement.
+     * This method is used before the statistics are updated to make sure nothing changes during the updating process.
+     */
+    @Override
+    public void lock( Statement statement ) {
```

Comment:
Because of circular dependencies it is not possible to use the LockManager directly in the statistics module. I could also only expose the LockManager and move the methods to the StatisticManager.
Before a reevaluation of the statistics, a lock has to be applied because otherwise it is possible that the entity changes during the creation of the algebra and it is not possible anymore to use the crated algebra.